lib/ddtrace.rb

```diff
@@ -11,23 +13,48 @@ module Datadog
   #   span = Datadog.tracer.trace('web.request')
   #   span.finish()
   #
+  # If you want to override the default tracer, the recommended way
+  # is to "pin" your own tracer onto your traced component:
+  #
+  #   tracer = Datadog::Tracer.new
+  #   pin = Datadog::Pin.get_from(mypatchcomponent)
+  #   pin.tracer = tracer
+
   def self.tracer
     @tracer
   end
 end
 
+begin
+  # We include 'redis-rails' here if it's available, doing it later
+  # (typically in initialize callback) does not work, it does not
+  # get loaded in the right context.
+  require 'redis-rails'
+  Datadog::Tracer.log.info("'redis-rails' module found, datadog redis integration is available")
+rescue LoadError
+  Datadog::Tracer.log.info("no 'redis-rails' module found, datadog redis integration is not available")
+end
+
+# Autopatching non-Rails contribs if env var is set
+Datadog::Monkey.autopatch_all
+
 # Datadog auto instrumentation for frameworks
 if defined?(Rails::VERSION)
   if Rails::VERSION::MAJOR.to_i >= 3
     require 'ddtrace/contrib/rails/framework'
 
+    Datadog::Monkey.patch_module(:redis) # does nothing if redis is not loaded
```

Comment:
There's IMHO no reason to do it for ES. For ES one should just manually call `Datadog::Monkey.patch_module(:elasticsearch)` as you would do in a non-rails context. The reason it's done here for Redis is that there's a specific way to integrate redis which is related to the fact Redis is often used as a cache backend.